bin/varnishd/cache/cache.h

```diff
@@ -789,20 +789,31 @@ void WS_ReleaseP(struct ws *ws, const char *ptr);
 void WS_Assert(const struct ws *ws);
 void WS_Reset(struct ws *ws, uintptr_t);
 void *WS_Alloc(struct ws *ws, unsigned bytes);
-void *WS_Copy(struct ws *ws, const void *str, int len);
+const void *WS_Copy(struct ws *ws, const void *str, int len);
 uintptr_t WS_Snapshot(struct ws *ws);
 int WS_Overflowed(const struct ws *ws);
-void *WS_Printf(struct ws *ws, const char *fmt, ...) v_printflike_(2, 3);
+const char *WS_Printf(struct ws *ws, const char *fmt, ...) v_printflike_(2, 3);
 int WS_Inside(const struct ws *, const void *, const void *);
 void WS_Assert_Allocated(const struct ws *ws, const void *ptr, ssize_t len);
+void WS_Id(const struct ws *ws, char *id);
 
 void WS_VSB_new(struct vsb *, struct ws *);
 char *WS_VSB_finish(struct vsb *, struct ws *, size_t *);
 
-static inline char*
-WS_Front(const struct ws *ws)
+static inline void *
+WS_Reservation(const struct ws *ws)
 {
-	return ws->f;
+
+	WS_Assert(ws);
+	return (ws->r != NULL ? ws->f : NULL);
+}
+
+static inline unsigned
+WS_Reserved(const struct ws *ws)
```

Comment:
can the names be better? `WS_Reserved()` could easily be misread as "is reserved" and `WS_Reaservation()` as "how big is the reservation".
What about `WS_ReservationP() ` and `WS_ReservationSize()`.

Reply:
I like `WS_Reaservation()` and `WS_ReservationSize()`.

Reply:
To clarify my choice, I went for "reserved" to have a function name not too long, and "reservation" was picked as a replacement for "front" (a noun).

Reply:
`WS_Reservation()` and `WS_ReservationSize()` sgtm